src/NLopt.jl

```diff
@@ -518,15 +518,15 @@ function optimize!(o::Opt, x::Vector{Cdouble})
     if length(x) != ndims(o)
         throw(BoundsError())
     end
-    opt_f = Array{Cdouble}(1)
+    opt_f = Array{Cdouble}(undef,1)
     ret = ccall((:nlopt_optimize,libnlopt), Cenum, (_Opt, Ptr{Cdouble},
                                                      Ptr{Cdouble}),
                 o, x, opt_f)
     return (opt_f[1], x, chks(ret))
 end
 
 optimize{T<:Real}(o::Opt, x::AbstractVector{T}) =
```

Comment:
use `AbstractVector{<:Real}` for this sort of thing

Reply:
OK